packages/cli/src/cmds/eval/eval-worker.ts

```diff
@@ -19,9 +18,18 @@ if (parentPort !== null) {
   parentPort.on('message', (message: { snapFilePath: string }) => {
     const { snapFilePath } = message;
 
-    new Compartment(getMockEndowments()).evaluate(
-      readFileSync(snapFilePath, 'utf8'),
-    );
+    const snapModule: { exports?: any } = { exports: {} };
+
+    new Compartment({
+      ...getMockEndowments(),
+      module: snapModule,
+      exports: snapModule.exports,
+    }).evaluate(readFileSync(snapFilePath, 'utf8'));
+
+    if (!snapModule.exports?.onMessage) {
```

Comment:
Should we hard fail if no message handler is registered or is there a use-case for it 🤔